src/NuGet.Clients/NuGet.CommandLine/Common/MSBuildProjectSystem.cs

```diff
@@ -413,7 +413,7 @@ private void RemoveEnsureImportedTarget(string targetsPath)
             }
 
             taskElement.Parent.RemoveChild(taskElement);
-            if (targetElement.Tasks.Count == 0)
+            if (((System.Collections.ICollection)targetElement.Tasks).Count == 0)
```

Comment:
When testing on my machine, it looks like `targetElement.Tasks` returns an instance of `Microsoft.Build.Collections.ReadOnlyCollection<T>` which is an internal type.
Could it be that C# dynamic can't deal with this out of the box?

I can't find much info on this and I'm definitely no expert but this thread suggest so http://stackoverflow.com/questions/15338429/c-sharp-dynamic-type-gotcha

Removing the cast breaks the test I added. So it might be interesting to see if you can also reproduce this?